### PR TITLE
EarthEnv landcover + WorldClim 2.1 + CHELSA V1

### LINKED ISSUE #20: Landcover dataset
https://www.earthenv.org/landcover

### LINKED ISSUE #22: Sliding window analysis
As an alternative to `coarsen`, it could be really good to implement a sliding window approach, which can be either using blocks, or using all pixels below a certain Haversine distance.

### LINKED ISSUE #23: Only load parts of the data
With **ArchGDAL** it is possible to only load a subset of the grid, not the entire grid. Because we know the total range, it should be possible to use something like

~~~ julia
datasource(layer; left=10.0, right=20.0, bottom=0.0, top=30.0)
~~~

to only get a subset. This would result in allocating a smaller array, which is definitely helpful when working with either `landcover` or `bioclim`.

Technically all of the pieces are already there since the package does this sort of operations when cropping.

### LINKED ISSUE #24: Loading parts of the datasets give different regions on different datasets
These three are supposedly the same latitudes/longitudes (after #23) , so this is concerning - **this is blocking #21**

Interestingly, the longitudes seem to be right, and it's the latitudes that get shifted. This suggests that the problem is with finding the part of the buffer to read in `geotiff`. Alternatively, `worldclim` seems to get the latitudes right, so maybe there is some weird GDAL thing going on with `bioclim` and `landcover`, which would require to dig into the documentation.

chelsa-heatmap
lc-heatmap
worldclim-heatmap

### LINKED ISSUE #25: Size of layers are inverted
According to @gabrieldansereau in #21 it looks like the size is inverted:

~~~ julia
julia> latitudes(temperature) |> length
2160

julia> longitudes(temperature) |> length
1080

julia> stride(temperature)
(0.16666666666666666, 0.04158950617283951)
~~~

In particular, the stride should have the same values. This is because *ArchGDAL*  can read things in a buffer with different dimensions, and this is in the process of being fixed.

### PR DESCRIPTION
**Pull request by tpoisot**
The new data have the same API as `worldclim`, but the files are much larger. In fact, the tests do not run as GitHub actions using the whole layer, and it takes a solid machine to run them at all. They work, but they can't be tested automatically -- as a quick check, I still run them using the new ability to only load parts of a dataset.

Because these new data do not always have a `NaN` type, the absence of data is now represented as `nothing` - this had all sort of consequences for almost everything in the package, none of which broke the API at all. But internally, this is a very extensive rewrite.

This PR also comes with improvements, specifically the sliding window analysis, the ability to broadcast over layers, and the ability to get a subset of the data from a file. The last point in particular makes a lot of data loading much faster.

- [x] earthenv - closes #20 
- [x] sliding window - closes #22 
- [x] subset of the data - closes #23 
- [x] correct buffer size for #23 - closes #25 
- [x] worldclim 2.1
- [x] export CHELSA `bioclim` function (high-res)
- [x] version bump (minor)
- [x] [updated documentation](https://ecojulia.github.io/SimpleSDMLayers.jl/previews/PR21/)
- [x] testing
- [x] broadcasting
- [x] fix the issue with latitudes when loading data subsets - closes #24

### CODE REVIEW

**Comment by tpoisot** [comment]
@gabrieldansereau Can you make sure that nothing in this PR is going to break what you are about to present at JuliaCon? This is a fairly big rewrite internally, but it didn't break any of the existing unit tests.

**Comment by gabrieldansereau** [comment]
I'll make sure it works while finishing my poster. I mean to show something similar to the examples in the documentation. As long as these are not broken, what I present shouldn't be either.

**Comment by gabrieldansereau** [comment]
@tpoisot Is this normal? The following
```
temperature = worldclim(1);
temperature_europe = temperature[left=-11.0, right=31.1, bottom=29.0, top=71.1];
size(temperature)
size(temperature_europe)
```

returns `(2160, 1080)` & `(507, 128)` with this version, but returns `(1080, 2160)` & `(254, 254)` on v0.2.2. Should the dimensions really be inversed in the first case and different in the second?

The `coarsen` example in the documentation is broken because of this.

**Comment by tpoisot** [comment]
@gabrieldansereau I have no idea why that's the case. I'll check this evening.

**Comment by tpoisot** [comment]
So, this is because the code of `geotiff` changed, and I would tempted to believe the new `geotiff` code, as using 

~~~ julia
worldclim(1; left=-11.0, right=31.1, bottom=29.0, top=71.1)
~~~

gives the correct result in terms of which area is covered. Let me check the size issue, it's weird.

**Comment by tpoisot** [comment]
Mmhhh. You're right @gabrieldansereau  , it looks like the size is inverted:

~~~ julia
julia> latitudes(temperature) |> length
2160

julia> longitudes(temperature) |> length
1080

julia> stride(temperature)
(0.16666666666666666, 0.04158950617283951)
~~~

In particular, the stride should have the same values - let me add a test to check this and then fix the bug.

**Comment by tpoisot** [comment]
**GREAT** now the github actions are not running. This is just _fantastic_. I'm ready to start a new life hoarding goats or something.

File: src/lib/overloads.jl [hunk 1]
```diff
@@ -240,7 +240,7 @@ the use of `copy`, manual update, and `convert`.
 function Base.similar(layer::T) where {T <: SimpleSDMLayer}
    emptygrid = similar(layer.grid)
    for i in eachindex(emptygrid)
-      emptygrid[i] = isnothing(layer.grid[i]) ? nothing : zero(eltype(layer.grid))
+      emptygrid[i] = isnothing(layer.grid[i]) ? nothing : zero(eltype(layer.grid[i]))
```
Comment (gabrieldansereau) [inline src/lib/overloads.jl:243]: `similar()` didn't work without this for climate layers & all. `eltype(temperature.grid)` is always `Union{Nothing, Float32}`, which has no method for `zero()`

**Pull request merged**